compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPUSelectUKernels.cpp

```diff
@@ -97,56 +140,39 @@ static ArrayAttr lookUpExecutableObjects(Operation *op) {
   return {};
 }
 
-/// Returns the function name and attributes to use for a ukernel with given
-/// `name` and `suffix` on the target described by `targetAttr`.
-static IREE::GPU::UKernelSpecAttr
-getUKernelSpec(StringRef name, StringRef suffix, MLIRContext *context,
-               IREE::HAL::ExecutableTargetAttr targetAttr) {
-  if (isROCMBackend(targetAttr)) {
-    auto nameAttr = StringAttr::get(
-        context, llvm::formatv("iree_uk_amdgpu_{}_{}", name, suffix));
-    auto defsAttr = DictionaryAttr::get(
-        context, {{StringAttr::get(context, "vm.import.module"),
-                   StringAttr::get(context, "rocm")}});
-    return IREE::GPU::UKernelSpecAttr::get(context, nameAttr, defsAttr);
+// Ensures that the op has ukernel bitcode as a hal.executable.object, stored
+// as a hal.executable.objects attribute on the op itself, ready to be hoisted
+// by the HoistExecutableObjects pass.
+// Returns failure if no bitcode was found for the configured ukernel.
+static LogicalResult
+ensureUKernelBitcode(Operation *op,
```

Comment:
nit: `validateUKernelBitcode`?

Reply:
I wanted to convey that this is mutating the op by doing ` op->setAttr` to provide the `hal.executable.objects`.  "Ensure" conveys that a bit, as in "ensure that an invariant has been satisfied". Can you think of a better verb?

Reply:
Ah, I missed this part. Seems fine as-is then.